### PR TITLE
Improve documentation

### PR DESCRIPTION
**Pull request by johan3141592**

### CODE REVIEW

File: internal/provider/validators.go [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+		return nil, []error{fmt.Errorf("failed to access file: %s", details)}
+	}
+
+	return nil, nil
+}
+
+// validateHash verifies that m contains a valid SHA-256 hash.
+func validateHash(m interface{}, p cty.Path) diag.Diagnostics {
+	if hash, ok := m.(string); ok && len(hash) == 64 {
```
[the remark this answers is not in the document]
Comment (leitzler) [reply internal/provider/validators.go:98]: Could be nice to have in the other places where sha256 is referred too.

File: internal/provider/resource_aws_private_container_registry.go [hunk 2]
```diff
@@ -31,28 +31,104 @@ import (
...
+mutation SetBundleApprovalStatus($input: SetBundleApprovalStatusInput!) {
+  setBundleApprovalStatus(input: $input)
+}
+´´´
+The input is an object with the following structure:
+´´´json
+{
+  "input": {
+    "approvalStatus": "APPROVED",
+    "bundleVersion": "1.164"
```
[the remark this answers is not in the document]
Comment (leitzler) [reply internal/provider/resource_aws_private_container_registry.go:92]: (same applies to multiple files in this diff)

**Review by leitzler — APPROVED** [review]
Just a few nits

File: docs/data-sources/aws_cnp_artifacts.md [hunk 3]
```diff
@@ -3,18 +3,123 @@
...
+  CLOUDNATIVEARCHIVAL_ENCRYPTION
+  BASIC - Represents the basic set of permissions required to onboard the feature.ENCRYPTION - Represents the set of permissions required for encryption operations.
+  CLOUDNATIVEPROTECTION
+  BASIC - Represents the basic set of permissions required to onboard the feature.EXPORT_AND_RESTORE - Represents the set of permissions required for export and
+  restore operations.FILE_LEVEL_RECOVERY - Represents the set of permissions required for file-level
+  recovery operations.SNAPSHOT_PRIVATE_ACCESS - Represents the set of permissions required for private
+  access to disk snapshots.
+  CLOUDNATIVES3_PROTECTION
+  BASIC - Represents the basic set of permissions required to onboard the feature.
+  EXOCOMPUTE\n" +
```
[the remark this answers is not in the document]
Comment (johan3141592) [reply docs/data-sources/aws_cnp_artifacts.md:22]: Thanks for finding this! I'll update the code accordingly.

File: internal/provider/resource_aws_private_container_registry.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (johan3141592) [reply internal/provider/resource_aws_private_container_registry.go:92]: Yep, and one where it shouldn't be one a bit down. Will fix it.

File: internal/provider/validators.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (johan3141592) [reply internal/provider/validators.go:98]: It's base 16 (Sprintf with `%x`). I will add it to the comment. What are the other places you are referring to?

[the remark this answers is not in the document]
Comment (leitzler) [reply internal/provider/validators.go:98]: No relevant other places as discussed offline 👍🏻

**Pull request merged**